### ISSUE TITLE
Progressions profile component returning incorrect and incomplete data for Seasonal Artifact

### ISSUE THREAD

**Issue by justrealmilk** [Bungie-net/api#1165]
As per title.

_Where once there was data now there is none_

profile.characterProgressions.data[member.characterId].seasonalArtifact:
```
{
  artifactHash: 2894222926     // last season's artifact
  pointsUsed: 0
  resetCount: 0
  tiers: []                    // should be populated
}
```

**Comment by vthornheart-bng** [comment]
Ooh, thank you for reporting this, I will definitely need to investigate what's going on there.

**Comment by vthornheart-bng** [comment]
TFS 855547

**Comment by justrealmilk** [comment]
The same applies to destiny2CoreSettings from the GetCommonSettings endpoint

**Comment by t3rr11** [comment]
I would like to also note, in characterVendors there is no vendor for the new Artifact, but the old one still exists. I followed the trail from the artifactHash to the vendors and noticed it. (Could be related)

**Comment by vthornheart-bng** [comment]
Yes, unfortunately you hit at the heart of this issue - I checked in a fix that should come down starting Tuesday, but here's the scoop:

To conserve on resources/try to stem the tide of content bloat that has been happening across seasons, they're trying to reuse as much season content as possible.  So that vendor is, I've been told, the single vendor that will be reused and its properties changed if/as needed for new seasons.

When I first made the fake "Artifact" definition that combines Vendor, Item, and some other data, I also was working under an understanding that the inventory item would also be reused.  That plan pivoted, and I totally failed to catch it.  As Abraham Lincoln would no doubt tell me, "now you fucked up!"

I'm changing how it's built to no longer assume the existence of a single shared item for all seasons' artifacts, and I'm adding some tests to try and catch this earlier should we mistakingly forget to do the manual configuration we need to do on our side in order to make sure the correct season's data comes down when the new season begins.

It will also come with an update to the documentation for currentSeasonalArtifactHash:

> The hash identifier of the DestinyArtifactDefinition mapped to the current season's artifact.  I've been told the artifact will be reused between seasons, but I'm building this because I am mashing on the 'doubt' button.
>  
> UPDATE: I was right to mash the doubt button.  Let it be recorded on this day, in the year of our Lord Two Thousand and Nineteen, on the Tenth of December.
